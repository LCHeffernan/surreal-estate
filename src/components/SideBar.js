import React from "react";
import { Link, useLocation } from "react-router-dom";
import qs from "qs";
import "../styles/sidebar.css";

const SideBar = () => {
  const buildQueryString = (operation, valueObj) => {
    const { search } = useLocation();
    const currentQueryParams = qs.parse(search, { ignoreQueryPrefix: true });
    const newQueryParams = {
      ...currentQueryParams,
      [operation]: JSON.stringify(valueObj),
    };
    return qs.stringify(newQueryParams, {
      addQueryPrefix: true,
      encode: false,
    });
  };
  return (
    <div className="side-bar">
      <div className="filter-city-title">Filter by city</div>
      <div className="city-link-container">
        <Link
          className="city-link"
          to={buildQueryString("query", { city: "Manchester" })}
        >
          Manchester
        </Link>
      </div>
      <div className="city-link-container">
        <Link
          className="city-link"
          to={buildQueryString("query", { city: "Leeds" })}
        >
          Leeds
        </Link>
      </div>
      <div className="city-link-container">
        <Link
          className="city-link"
          to={buildQueryString("query", { city: "Sheffield" })}
        >
          Sheffield
        </Link>
      </div>
      <div className="city-link-container">
        <Link
          className="city-link"
          to={buildQueryString("query", { city: "Liverpool" })}
        >
          Liverpool
        </Link>
      </div>
      <div className="sort-city-title">Sort By</div>
      <div className="sort-link-container">
        <Link className="sort-link" to={buildQueryString("sort", { price: 1 })}>
          Price Ascending
        </Link>
      </div>
      <div className="sort-link-container">
        <Link
          className="sort-link"
          to={buildQueryString("sort", { price: -1 })}
        >
          Price Descending
        </Link>
      </div>
    </div>
  );
};

export default SideBar;
