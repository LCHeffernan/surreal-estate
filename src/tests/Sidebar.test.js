import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import SideBar from "../components/SideBar";

describe("Sidebar", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <SideBar />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders title text", () => {
    const { getByText } = render(
      <BrowserRouter>
        <SideBar />
      </BrowserRouter>
    );
    expect(getByText(/filter by city/i)).toBeInTheDocument();
  });
});
