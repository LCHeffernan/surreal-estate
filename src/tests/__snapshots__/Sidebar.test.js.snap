// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="side-bar"
  >
    <form>
      <input
        class="search-input"
        type="text"
        value=""
      />
      <button
        class="search-button"
        type="submit"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-magnifying-glass "
          data-icon="magnifying-glass"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352c79.5 0 144-64.5 144-144s-64.5-144-144-144S64 128.5 64 208s64.5 144 144 144z"
            fill="currentColor"
          />
        </svg>
      </button>
    </form>
    <div
      class="filter-city-title"
    >
      Filter by city
    </div>
    <div
      class="city-link-container"
    >
      <a
        class="city-link"
        href="/?query={\\"city\\":\\"Manchester\\"}"
      >
        Manchester
      </a>
    </div>
    <div
      class="city-link-container"
    >
      <a
        class="city-link"
        href="/?query={\\"city\\":\\"Leeds\\"}"
      >
        Leeds
      </a>
    </div>
    <div
      class="city-link-container"
    >
      <a
        class="city-link"
        href="/?query={\\"city\\":\\"Sheffield\\"}"
      >
        Sheffield
      </a>
    </div>
    <div
      class="city-link-container"
    >
      <a
        class="city-link"
        href="/?query={\\"city\\":\\"Liverpool\\"}"
      >
        Liverpool
      </a>
    </div>
    <div
      class="sort-city-title"
    >
      Sort By
    </div>
    <div
      class="sort-link-container"
    >
      <a
        class="sort-link"
        href="/?sort={\\"price\\":1}"
      >
        Price Ascending
      </a>
    </div>
    <div
      class="sort-link-container"
    >
      <a
        class="sort-link"
        href="/?sort={\\"price\\":-1}"
      >
        Price Descending
      </a>
    </div>
  </div>
</DocumentFragment>
`;
